integration-tests/test_ui.py

```diff
@@ -42,7 +42,12 @@ async def local_hub_local_binder(request):
                 break
         except requests.exceptions.ConnectionError:
             pass
-        await asyncio.sleep(1)
+        # We use a increased linear time between requests
+        # because BinderHub server might require more than 10s to start.
+        time.sleep(i)
+    else:
+        print("BinderHub server is down! Tests will probably fail.")
```

Comment:
If we move the `sleep` to the start then we're certain the server is returning 404, so we can fail here for clarity.